### ISSUE TITLE
cilium monitor logs identity of world as host for replies

### ISSUE THREAD

**Issue by joestringer** [cilium/cilium#3339]
I was running a container and pinging a public IP (8.8.8.8), and here is the `cilium monitor` output:

```
# cilium monitor
Listening for events on 2 CPUs with 64x4096 of shared memory   
Press Ctrl-C to quit           
<- endpoint 42910 flow 0x6684bcc7 identity 23405->0 state new ifindex 0: 10.15.236.254 -> 8.8.8.8 EchoRequest                  
-> stack flow 0x6684bcc7 identity 23405->2 state established ifindex 0: 10.15.236.254 -> 8.8.8.8 EchoRequest                   
<- host flow 0x6684bcc7 identity 1->0 state new ifindex eth0: 8.8.8.8 -> 10.15.236.254 EchoReply
-> endpoint 42910 flow 0x6684bcc7 identity 1->23405 state reply ifindex lxc2bde3: 8.8.8.8 -> 10.15.236.254 EchoReply
```

The Request is sent from the endpoint to id 0 (unknown), which is then sent to the stack (at which point we know it is for the world, ie identity 2). When the reply comes back, the identity appears as 1 (host), not 2(world).

**Comment by joestringer** [comment]
In the datapath, we could do some simple logic like:
```
srcID = WORLD_ID;
if (!extract_id(skb, &srcID)) {
    if (ip_is_host(src))
        srcID = HOST_ID;
    else if (ip_is_in_cluster(src))
        srcID = CLUSTER_ID;
}
```

**Comment by joestringer** [comment]
Fixed by #3988.
